packages/api-client/src/components/Server/ServerVariablesForm.vue

```diff
@@ -0,0 +1,64 @@
+<script setup lang="ts">
+import ServerVariablesSelect from '@/components/Server/ServerVariablesSelect.vue'
+import ServerVariablesTextbox from '@/components/Server/ServerVariablesTextbox.vue'
+import type {
+  ServerVariableValues,
+  ServerVariables,
+} from '@/components/Server/types'
+import { nanoid } from 'nanoid'
+
+const props = defineProps<{
+  variables?: ServerVariables
+  values?: ServerVariableValues
+  /** The ID of the input controlled by the variables form */
+  controls?: string
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:variable', name: string, value: string): void
+}>()
+
+const baseId = nanoid()
+
+function setVariable(name: string, value: string) {
+  emit('update:variable', name, value)
+}
+
+const getVariable = (name: string) => {
+  return (
+    props.values?.[name] ??
+    props.variables?.[name]?.default ??
+    ''
+  ).toString()
+}
+</script>
+<template>
+  <template v-if="variables && Object.keys(variables ?? {}).length">
+    <template
+      v-for="name in Object.keys(variables)"
+      :key="name">
+      <div class="group flex w-full">
+        <label
+          class="flex items-center pl-3 py-1.5 after:content-[':'] mr-1.5 group-has-[input]:mr-0"
+          :for="`${baseId}-variable-${name}`">
```

Comment:
It looks like if the text box doesn't exist this `for` is broken? 

Rather than use an id here you can just make the label element the wrapper element (e.g. `<div class="group flex w-full">` → `<label class="group flex w-full">` and turn the label into a div or span) and it should automatically associate for you

Reply:
absolutely right will do! just copy pasted from reference so far